### PR TITLE
Get invitations v2

### PR DESCRIPTION
**Pull request by siuwafong**
I made some changes to the mongoose schemas, added an error utility file in the utils folder, and added the get request for invitations.

### CODE REVIEW

File: server/models/messages.js [hunk 1]
```diff
@@ -13,7 +12,10 @@ const MessageSchema = new Schema({
         ref: "User",
         required: true
     },
-    type: String,
+    type: {
+        type: String,
+        enum: ["msg", "img"]
```
Comment (sina-jamshidi) [inline server/models/messages.js:17]: 👍

File: server/models/invites.js [hunk 2]
```diff
@@ -12,7 +11,10 @@ const InviteSchema = new Schema({
         ref: "User",
         required: true
     },
-    status: String,
+    status: {
+        type: String,
+        enum: ["pending", "declined", "accepted"]
```
Comment (sina-jamshidi) [inline server/models/invites.js:16]: Nice addition 👍

File: server/routes/index.js [hunk 3]
```diff
@@ -2,14 +2,22 @@ const express = require("express");
...
 });
 
-router.get("/user/:id/invitations", (req, res, next) => {
-  // TODO
-  
+router.get("/user/:id/invitations", async (req, res, next) => {
+  const { id } = req.params
+  try {
+    const invites = await Invite.find({recipient: id}).populate('sender')
+    res.json(invites)
```
Comment (sina-jamshidi) [inline server/routes/index.js:16]: one question to consider here would be, do we want the sender to also see the invites that they have sent out?

File: server/schemas.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const Joi = require('joi')
+
```
Comment (sina-jamshidi) [inline server/schemas.js:2]: Nice validation

**Review by sina-jamshidi — APPROVED** [review]
Looks good to merge!

File: server/routes/index.js (hunk 3 above)
Comment (siuwafong) [reply server/routes/index.js:16]: Good idea. I will try to make some changes to include that.

**Pull request merged**